Merged with upstream PUP 4.9.0
(PUP-6792) SMF should recursively enable services
https://tickets.puppetlabs.com/browse/PUP-6792

diff --git a/lib/puppet/provider/service/smf.rb b/lib/puppet/provider/service/smf.rb
index be04a95..d006ca7 100644
--- a/lib/puppet/provider/service/smf.rb
+++ b/lib/puppet/provider/service/smf.rb
@@ -80,7 +80,7 @@ Puppet::Type.type(:service).provide :smf, :parent => :base do
     when :maintenance, :degraded
       [command(:adm), :clear, @resource[:name]]
     else
-      [command(:adm), :enable, "-s", @resource[:name]]
+      [command(:adm), :enable, "-rs", @resource[:name]]
     end
   end
 
